internal/core/metadata/operators/command/get_test.go

```diff
@@ -48,9 +66,27 @@ func TestLoadAllCommands(t *testing.T) {
 		expectedResult []models.Command
 		expectError    bool
 	}{
-		{"GetAllOK", config.ServiceInfo{MaxResultCount: 1}, createAllCommandsLoaderMock(), []models.Command{testCommand}, false},
-		{"GetAllFailMaxExceeded", config.ServiceInfo{}, createAllCommandsLoaderMock(), []models.Command{testCommand}, true},
-		{"GetAllFailUnexpected", config.ServiceInfo{}, createAllCommandsLoaderMockFail(), nil, true},
+		{
+			"GetAllOK",
+			config.ServiceInfo{MaxResultCount: 1},
+			createCommandLoaderMock("GetAllCommands", []models.Command{testCommand}, nil, ""),
+			[]models.Command{testCommand},
+			false,
+		},
+		{
+			"GetAllFailMaxExceeded",
+			config.ServiceInfo{},
+			createCommandLoaderMock("GetAllCommands", []models.Command{testCommand}, nil, ""),
+			[]models.Command{testCommand},
```

Comment:
I know this isn't what this PR is about, but in the max result count exceeded case the operator should return `[]models.Command{}, errors.NewErrLimitExceeded(op.config.MaxResultCount)`, and not return the Commands returned by the database.

Reply:
I do not agree totally because createCommandLoaderMock define how the mocked function GetAllCommands should behave. 
[GetAllCommands](https://github.com/edgexfoundry/edgex-go/blob/aedf85076467d5f6e837d3bd8b59809f5e360a0f/internal/core/metadata/operators/command/get.go#L44) does not return NewErrLimitExceeded, the error is returned one layer [above](https://github.com/edgexfoundry/edgex-go/blob/aedf85076467d5f6e837d3bd8b59809f5e360a0f/internal/core/metadata/operators/command/get.go#L50) in the [commandLoadAll.Execute](https://github.com/edgexfoundry/edgex-go/blob/aedf85076467d5f6e837d3bd8b59809f5e360a0f/internal/core/metadata/operators/command/get.go#L43). 
But thank you for your comment, it help me realize that my return statement in case of NewErrLimitExceeded is wrong. (I am suppose to return empty list in case of NewErrLimitExceeded)

Reply:
I think we were in agreement :)

I didn't mean the mock db command should return that error--but that the Execute function should return `[]models.Command{}, errors.NewErrLimitExceeded(op.config.MaxResultCount)` and that the test should be expect that.